pkg/protobuf/comparer_test.go

```diff
@@ -0,0 +1,532 @@
+		"new definition does not contain the message 'first'",
+	}, problems)
+}
+
+func Test_Comparer_WithFieldRepeatedChange(t *testing.T) {
+	current := `syntax = "proto3";
+	package gen;
+
+	message first {
+		repeated string one = 1;
+	}
+
+	message record {
+		first first_message = 1;
+	}
+	`
+
+	new := `syntax = "proto3";
+	package gen;
+
+	message first {
+		string one = 1;
+	}
+
+	message record {
+		first first_message = 1;
+	}
+	`
+
+	ok, problems, err := protobuf.CheckForBreakingChanges([]byte(current), []byte(new))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, false, ok)
+	assert.DeepEqual(t, []string{
+		"message 'first' had field repeated 'true' changed to 'false' at sequence '1'",
+	}, problems)
+}
+
+func Test_Comparer_WithEnumChangedValue(t *testing.T) {
+	current := `syntax = "proto3";
+	package gen;
+
+	enum first {
+		ZERO = 0;
+		ONE = 1;
+		TWO = 2;
+	}
+
+	message record {
+		first first_message = 1;
+	}
+	`
+
+	new := `syntax = "proto3";
+	package gen;
+
+	enum first {
+		WHOOPS = 0;
+		ONE = 1;
+		TWO = 2;
+	}
+
+	message record {
+		first first_message = 1;
+	}
+	`
+
+	ok, problems, err := protobuf.CheckForBreakingChanges([]byte(current), []byte(new))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, false, ok)
+	assert.DeepEqual(t, []string{
+		"enum 'first' had value changed from '{ZERO}' to '{WHOOPS}' at sequence '0'",
+	}, problems)
+}
+
+func Test_Comparer_WithEnumMissingValue(t *testing.T) {
+	current := `syntax = "proto3";
+	package gen;
+
+	enum first {
+		ZERO = 0;
+		ONE = 1;
+		TWO = 2;
+	}
+
+	message record {
+		first first_message = 1;
+	}
+	`
+
+	new := `syntax = "proto3";
+	package gen;
+
+	enum first {
+		ONE = 1;
+		TWO = 2;
+	}
+
+	message record {
+		first first_message = 1;
+	}
+	`
+
+	ok, problems, err := protobuf.CheckForBreakingChanges([]byte(current), []byte(new))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, false, ok)
+	assert.DeepEqual(t, []string{
+		"enum 'first' does not contain value sequence '0' with name '{ZERO}'",
+	}, problems)
+}
+
+func Test_Comparer_WithMissingEnum(t *testing.T) {
+	current := `syntax = "proto3";
+	package gen;
+
+	enum first {
+		ZERO = 0;
+		ONE = 1;
+		TWO = 2;
+	}
+
+	message record {
+		first first_message = 1;
+	}
+	`
+
+	new := `syntax = "proto3";
+	package gen;
+
+	enum last {
+		ZERO = 0;
+		ONE = 1;
+		TWO = 2;
+	}
+
+	message record {
+		first first_message = 1;
+	}
+	`
+
+	ok, problems, err := protobuf.CheckForBreakingChanges([]byte(current), []byte(new))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, false, ok)
+	assert.DeepEqual(t, []string{
+		"new definition does not contain enum 'first'",
+	}, problems)
+}
+
+func Test_Comparer_WithMissingEmbeddedEnum(t *testing.T) {
+	current := `syntax = "proto3";
+	package gen;
+
+	message first {
+		enum enum1 {
+			ZERO = 0;
+			ONE = 1;
+			TWO = 2;
+		}
+
+		enum1 enum_value = 1;
+	}
+
+	message record {
+		first first_message = 1;
+	}
+	`
+
+	new := `syntax = "proto3";
```

Comment:
Is this a valid proto? Seems like enum1 would need to exist somewhere to validate.

Reply:
It is not. It wouldn't compile. Just making sure to catch the case since we won't lint/compile the protos